src/main/c/com/micromouse/robot/robot.c

```diff
@@ -182,4 +182,60 @@ void RobotDestructor(Robot* winslow) {
 
 	// Free up the memory for the Robot
 	free(winslow);
-}
\ No newline at end of file
+}
+
+// Robot Location -> [Linked-list of [Moves]]
+// Returns a linked list of moves for the robot to execute, starts by 
+// searching the maze for the path to the goal, and returns the path as a 
+// Linked-list of Moves.
+Cell* SolveMaze(Robot* winslow, Location *goal) {
+  // Create path to return and load starting location into path
+  Location* curr_loc = winslow->location_;
+
+  // Mark curr_loc as visited
+  int curr_x = curr_loc->x;
+  int curr_y = curr_loc->y;
+  List* queue = InitializeList(winslow->maze_[curr_x][curr_y]);
+  Cell* curr_cell = winslow->maze_[curr_x][curr_y];
+  curr_cell->visited = 1;
+
+  int is_queue_empty = empty(queue);
+  while(!is_queue_empty) {
+    // TODO: Dequeue curr_cell from queue.
+    curr_cell = Front(&queue);
+    if(curr_cell->location == goal) {
+      return winslow->maze_[goal->x][goal->y];
+    }
+    curr_loc = curr_cell->location;
+    int x_ = curr_loc->x;
+    int y_ = curr_loc->y;
+    if(curr_cell->east) {
+      queue = VisitNeighbor(curr_cell, x_ + 1, y_, winslow, queue);
+    }
+    if(curr_cell->west) {
+      queue = VisitNeighbor(curr_cell, x_ - 1, y_, winslow, queue);
+    }
+    if(curr_cell->north) {
+      queue = VisitNeighbor(curr_cell, x_, y_ + 1, winslow, queue);
+    }
+    if(curr_cell->south) {
+      queue = VisitNeighbor(curr_cell, x_, y_ - 1, winslow, queue);
+    }
+    is_queue_empty = empty(queue);
+  }
+  return winslow->maze_[0][0]; // Did not find goal cell, return fail.
+}
+
+// Visit neighbor and update the queue
+List* VisitNeighbor(Cell* current_cell, int x, int y, Robot* winslow, List* queue) {
+  Cell* neighbor_cell = winslow->maze_[x][y];
+  if(!neighbor_cell->visited) {
+    neighbor_cell->visited = 1;
+    neighbor_cell->parent = current_cell;
+    Location* neighbor_loc = neighbor_cell->location;
+    Append(winslow->maze_[neighbor_loc->x][neighbor_loc->y], &queue);
```

Comment:
So you are probably passing in the address of the queue so that the types didnt complain but that is not what you want to do here. You need to pass in a pointer to the front of the list because you have to modify the first element of the list